internal/workers/reportwkr/reportwkr.go

```diff
@@ -0,0 +1,50 @@
+package reportwkr
+
+import (
+	"context"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/penguin-statistics/backend-next/internal/config"
+	"github.com/penguin-statistics/backend-next/internal/service"
+	"go.uber.org/fx"
+	"runtime"
+)
+
+type ReportConsumerDeps struct {
+	fx.In
+	ReportServices *service.Report
+}
+
+type ReportConsumerWorker struct {
+	// count is the number of workers
+	numWorker int
+
+	ReportConsumerDeps
+}
+
+func Start(conf *config.Config, deps ReportConsumerDeps) {
```

Comment:
Would `func (w ReportConsumerWorker) Start(conf *config.Config)` work? Not sure if `fx` will properly initialize that for us but on my head I think it might be able to do so. If not then this is fine.

Reply:
@GalvinGao It would work. 

But notice the trap that `ReportConsumerWorker` "extends" (specifically, promotes the fields in) `ReportConsumerDeps`, and `ReportConsumerDeps` extends `fx.In`. Hence `ReportConsumerWorker` extends `fx.In`. 

So `ReportConsumerWorker` cannot be instantiated automatically. It would cause something like `unexported fields numWorker not allowed in fx.In`.